packages/@lwc/engine-core/src/framework/base-bridge-element.ts

```diff
@@ -121,9 +142,25 @@ export function HTMLBridgeElementFactory(
     const { attributeChangedCallback: superAttributeChangedCallback } = SuperClass.prototype as any;
     const { observedAttributes: superObservedAttributes = [] } = SuperClass as any;
     const descriptors: PropertyDescriptorMap = create(null);
+
+    // present a hint message so that developers are aware that they have not decorated property with @api
+    if (process.env.NODE_ENV !== 'production') {
+        for (const propName in protoDescriptors) {
```

Comment:
```suggestion
        for (const propName of keys(protoDescriptors)) {
```

`for...in` is generally discouraged these days, although I admit in this case it doesn't matter since there's no problem with `hasOwnProperty` here.